src/irc/core/massjoin.c

```diff
@@ -39,12 +39,15 @@ static void event_join(IRC_SERVER_REC *server, const char *data,
 	IRC_CHANNEL_REC *chanrec;
 	NICK_REC *nickrec;
 	GSList *nicks, *tmp;
+	gboolean send_massjoin;
 
 	g_return_if_fail(data != NULL);
 
 	if (g_ascii_strcasecmp(nick, server->nick) == 0) {
-		/* You joined, no need to do anything here */
-		return;
+		/* You joined, do not massjoin */
+		send_massjoin = FALSE;
+	} else {
+		send_massjoin = TRUE;
 	}
```

Comment:
Hmm, can you explain this part? The return not being here means we call irc_nicklist_insert, which we previously didn't at this point. What is this for?

I think I understand the massjoin part but don't care much about it.

Reply:
The goal is that we fill in our OWN account here as well, if we receive it. Otherwise we would need more special code for that

Reply:
should I expand the comment?

Reply:
I think the comment is fine for the code, I'm just wondering about the diff. So it's fine to just do the irc_nicklist_insert here? I guess something else might do that later with fewer details and end up as a no-op?

Reply:
normally our own nick gets inserted only as part of `event_names_list` in irc-nicklist.c when the NAMES are received after join (with a special case further down to insert our own nick in any case if it was NOT contained in NAMES). The code in massjoin.c handles all the JOINs that happen after we are joined to a channel.

When I relax that check, the (special?) self-JOIN that we get when joining a channel will also be fed through the JOIN code in massjoin.c, with the benefit that we can leech on the extended-account there.

As a consequence, I had to change irc-nicklist.c's NAMES handler to update our CHANNEL USER MODES from the NAMES.

I don't know if all of this is a very bad(tm) idea and if it would be better to do it in some other way.

What do you think? Can you follow my explanation?